ivy/functional/backends/tensorflow/elementwise.py

```diff
@@ -19,8 +19,8 @@ def bitwise_left_shift(
 
 
 def add(
-    x1: Union[tf.Tensor, tf.Variable],
-    x2: Union[tf.Tensor, tf.Variable],
+    x1: Union[float, tf.Tensor, tf.Variable], 
+    x2: Union[float, tf.Tensor, tf.Variable]
 ) -> Union[tf.Tensor, tf.Variable]:
     if hasattr(x1, "dtype") and hasattr(x2, "dtype"):
         promoted_type = tf.experimental.numpy.promote_types(x1.dtype, x2.dtype)
```

Comment:
the line `x1 = tf.constant(x1, dtype=x2.dtype)` below will throw an error if `x2` is not an array. Please check for these kinds of issues for all backends.